tarofreighter/chain_porter.go

```diff
@@ -520,11 +560,18 @@ func (p *ChainPorter) advanceState(currentPkg *sendPackage,
 		if sendPkg.SendState == SendStateStoreProofs {
 			// At this point, the transfer transaction is confirmed
 			// on-chain. We go on to store the sender and receiver
-			// proofs in the proof archive and then attempt to
-			// deliver the receiver proof.
+			// proofs in the proof archive.
 			logState(sendPkg.SendState)
-			p.storeAndTransferProofs(sendPkg)
+			p.storeProofs(sendPkg)
 		}
+
+		if sendPkg.SendState == SendStateReceiverProofTransfer {
```

Comment:
See prior comment about folding these states above into the main switch statement of the state machine.